core/src/server.rs

```diff
@@ -56,11 +57,26 @@ pub async fn run_on_port(config: Config, port: u16) {
     run(config, Some(port), None).await
 }
 
+#[cfg(unix)]
+async fn exit_server() {
+    use nix::sys::signal;
+    use nix::unistd::Pid;
+
+    // The server has a graceful handler for SIGHUP.
+    _ = nix::sys::signal::kill(Pid::this(), signal::Signal::SIGHUP);
+}
```

Comment:
Does serial-keel exit and stay killed, or does the serial-keel service try to restart it? Also should this return something to confirm the kill?

Reply:
This feature does not know anything about whether serial-keel is managed via the user or via a service, so that's up to the service.
If the service feature we merged to SK is used, that has a restart policy, so I assume that would just restart the server.


This feature is more geared towards where `systemd` is not used.
Else we could just `systemctl restart / stop`.

Reply:
Return code: I'm not sure how that could be handled gracefully, because there is likely a race condition between sending the return and the server being closed.

Reply:
alright, if the server was gone, we'd know anyway by the endpoints not working.